src/common/system-ca.ts

```diff
@@ -20,22 +20,93 @@
 
-    logger.debug(`Using host CA: ${attributes.join(",")}`);
+export function isCertActive (cert: string) {
+  const isExpired = typeof cert !== "string" || cert.includes(DSTRootCAX3);
+
+  return !isExpired;
+}
+
+/**
+ * Get root CA certificate from MacOSX system keychain
+ * Only return non-expred certificates.
+ */
+export async function getMacRootCA () {
+  // inspired mac-ca https://github.com/jfromaniello/mac-ca
+  const args = "find-certificate -a -p";
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Cheatsheet#other_assertions
+  const splitPattern = /(?=-----BEGIN\sCERTIFICATE-----)/g;
+  const systemRootCertsPath = "/System/Library/Keychains/SystemRootCertificates.keychain";
+  const bin = "/usr/bin/security";
+  const trusted = (await promiseExec(`${bin} ${args}`)).stdout.toString().split(splitPattern);
+  const rootCA = (await promiseExec(`${bin} ${args} ${systemRootCertsPath}`)).stdout.toString().split(splitPattern);
+
+  return [...new Set([...trusted, ...rootCA])].filter(isCertActive);
+}
+
+/**
+ * Get root CA certificate from Windows system certificate store.
+ * Only return non-expred certificates.
+ */
+export function getWinRootCA (): Promise<string[]> {
+  return new Promise((resolve) => {
+    const CAs: string[] = [];
+
+    wincaAPI({
+      format: wincaAPI.der2.pem,
+      inject: false,
+      ondata: (ca: string) => {
+        CAs.push(ca);
+      },
+      onend: () => {
+        resolve(CAs.filter(isCertActive));
+      }
+    });
+  });
+}
+
+
+/**
+ * Add (or merge) CAs to https.globalAgent.options.ca
+ */
+export function injectCAs (CAs: string[]) {
+  for (const cert of CAs) {
+    if (Array.isArray(https.globalAgent.options.ca) && !https.globalAgent.options.ca.includes(cert)) {
+      https.globalAgent.options.ca.push(cert);
+    } else {
+      https.globalAgent.options.ca = [cert];
+    }
   }
```

Comment:
I think this might be clearer if this was the solution:

```ts
  const oldCAs = [https.globalAgent.options.ca].flat();
  const injectedCAs = new Set([...oldCAs, ...CAs]);

  https.globalAgent.options.ca = [...injectedCAs];
```

Reply:
Ok, sounds good

Reply:
Changed https://github.com/lensapp/lens/pull/4185/commits/ed7d9f4541df0a7995f8e7027e9c7a82687c771a

Reply:
This doesn't work as expected, getting this, I will revert.
```
[0]   rawException: TypeError [ERR_INVALID_ARG_TYPE]: The "options.ca" property must be of type string or an instance of Buffer, TypedArray, or DataView. Received undefined
[0]       at validateKeyOrCertOption (_tls_common.js:83:11)
[0]       at Object.createSecureContext (_tls_common.js:109:9)
[0]       at Object.connect (_tls_wrap.js:1580:48)
[0]       at Agent.createConnection (https.js:129:22)
[0]       at Agent.createSocket (_http_agent.js:323:26)
[0]       at Agent.addRequest (_http_agent.js:274:10)
[0]       at new ClientRequest (_http_client.js:306:16)
[0]       at Object.request (https.js:313:10)
[0]       at y._performRequest (/Users/hchen/.k8slens/extensions/lenscloud-lens-extension/dist/main.js:15:12465)
[0]       at new y (/Users/hchen/.k8slens/extensions/lenscloud-lens-extension/dist/main.js:15:8632) {
[0]     code: 'ERR_INVALID_ARG_TYPE'
[0]   }
[0] }
```

Reply:
I guess `https.globalAgent.options.ca` is a TypedArray, the else may have bugs though..